common/lib/main.s

```diff
@@ -0,0 +1,20 @@
+.text
+.global __main
+.global __exit
+
+__main:
+    TSX
+    STX __save_sp
+    JMP main
+
+__exit:
+   STX __save_x
```

Comment:
`STX mos8(__rc2)`. Since __exit is a normal C function, the scratch zero page registers are available.